src/types/json.h

```diff
@@ -215,6 +215,32 @@ struct JsonValue {
     return Status::OK();
   }
 
+  StatusOr<std::vector<std::optional<JsonValue>>> ArrPop(std::string_view path, int64_t index = -1) {
+    std::vector<std::optional<JsonValue>> popped_values;
+
+    try {
+      jsoncons::jsonpath::json_replace(value, path,
+                                       [&popped_values, index](const std::string & /*path*/, jsoncons::json &val) {
+                                         if (val.is_array() && !val.empty()) {
+                                           auto popped_iter = val.array_range().begin();
+                                           if (index == -1 || index >= static_cast<int64_t>(val.size())) {
+                                             popped_iter = val.array_range().end() - 1;
+                                           } else if (index > 0) {
+                                             popped_iter += index;
+                                           }
```

Comment:
I think if the index is negative, we need to calculate the real position from the end. 
```shell
127.0.0.1:6666> json.set a $ [1,2,3,4]
OK
127.0.0.1:6666> json.arrpop a $ -2
1) "3"
127.0.0.1:6666>
```
I recommend you to try it by yourself : )

Reply:
I'm sorry I misunderstood `Out-of-range indexes round to their respective array ends` stated in the doc.

I've fixed it based on the below code in RedisJson.

```rust
impl ArrayIndex for i64 {
    fn normalize(self, len: i64) -> usize {
        let index = if self < 0 {
            len - len.min(-self)
        } else if len > 0 {
            (len - 1).min(self)
        } else {
            0
        };
        index as usize
    }
}
```